src/components/keyboard/virtual-keyboard.ts

```diff
@@ -0,0 +1,76 @@
+import type { Cell } from "@/components/GameCell.vue";
+import { reactive, watch } from "vue";
+import { useGrid } from "@/grid";
+
+export type HintType = "NO_INFORMATION" | "ABSENT" | "CORRECT" | "MISPLACED";
+
+export interface VirtualKeyboardLine {
+  keys: VirtualKeyboardKey[];
+}
+
+export interface VirtualKeyboardKey {
+  value: string;
+  letterHint?: HintType;
+}
+
+const { grid, currentLine, handleLetter, handleEnter, handleBackspace } =
+  useGrid();
+
+const firstLineKeys = ["A", "Z", "E", "R", "T", "Y", "U", "I", "O", "P"];
+const secondLineKeys = ["Q", "S", "D", "F", "G", "H", "J", "K", "L", "M"];
+const thirdLineKeys = ["W", "X", "C", "V", "B", "N"];
+
+const keyboardLines: VirtualKeyboardLine[] = reactive(
+  [firstLineKeys, secondLineKeys, thirdLineKeys].map((lineKeys) => ({
+    keys: lineKeys.map(
+      (letter) =>
+        ({
+          value: letter,
+          letterHint: "NO_INFORMATION",
+        } as VirtualKeyboardKey)
+    ),
+  }))
+);
+
+watch(currentLine, (currentLine) => {
+  displayHints(currentLine - 1);
+});
+
+export function useVirtualKeyboard() {
+  function handleVirtualKey(key: VirtualKeyboardKey) {
+    if (isCharacter(key.value)) handleLetter(key.value);
+    else if (key.value === "↲") handleEnter();
+    else if (key.value === "⌫") handleBackspace();
+  }
+
+  return { keyboardLines, handleVirtualKey };
+}
+
+// TODO sortir dans un composable "utilitaire"
+function isCharacter(keyValue: string) {
+  return /^[A-Z]$/.test(keyValue.toUpperCase());
+}
+
+function displayHints(currentLine: number): void {
```

Comment:
Warning sonarlint `'currentLine' is already declared in the upper scope.` ainsi que pour `key` de `keys.find((key)`

Reply:
corrigé